tests/test_init.py

```diff
@@ -0,0 +1,356 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from contextlib import contextmanager
+from pathlib import Path
+
+import pytest
```

Comment:
Can we include a test that demonstrates the ability to pull the name of the directory we created, as well as converting it into an identifier? 

See https://github.com/snowflakedb/snowflake-cli/blob/main/src/snowflake/cli/plugins/nativeapp/init.py#L323 and the lines after it -- we allow the user to set the project name variable if desired, but if it's not provided we pull it from the path name and turn it into an identifier with a set of rules that prevents users from hitting rough edges (e.g we'll convert `-` to `_` to prevent users from accidentally opting into quoted identifiers).

Reply:
The identifier is not a case for some projects - for example for Snowpark, where we have bunch of different objects defined in one `snowflake.yml`

This particular feature (as well as `--name` flag) will be preserved on `snow app init` side.

Reply:
I've created `to_snowflake_identifier` jinja filter, which can be used in templates to convert strings into Snowflake identifiers